yggdrasil_decision_forests/learner/decision_tree/oblique.cc

```diff
@@ -788,12 +788,31 @@ void SampleProjection(const absl::Span<const int>& features,
     }
   };
 
+  #ifndef NDEBUG  // Keep DCHECK_EQ from for feature : features
   for (const auto feature : features) {
     DCHECK_EQ(data_spec.columns(feature).type(), dataset::proto::NUMERICAL);
-    if (unif01(*random) < projection_density) {
-      projection->push_back({feature, gen_weight(feature)});
-    }
   }
+  #endif
+
+  const size_t p = features.size();
```

Comment:
Same as the "k" comment, I would rename this "num_features" or have a separate function.

Alternatively, since this variable is only used twice, it might be worth using "features.size()" directly.